### PR TITLE
test(react-drawer): add e2e for base drawer components

### PR DESCRIPTION
**Pull request by marcosmoura**
Add missing Cypress tests for Drawer, DrawerInline and DrawerOverlay components.

### CODE REVIEW

File: packages/react-components/react-drawer/.eslintrc.json [hunk 1]
```diff
@@ -1,4 +1,13 @@
 {
   "extends": ["plugin:@fluentui/eslint-plugin/react"],
-  "root": true
+  "root": true,
+  "overrides": [
```
Comment (marcosmoura) [inline packages/react-components/react-drawer/.eslintrc.json:4]: Override to allow us to have shared tests to be placed on the e2e directory. This allows us to create a function encapsulating base scenarios that can be called for the Drawer, DrawerInline and DrawerOverlay.

File: packages/react-components/react-drawer/tsconfig.cy.json [hunk 2]
```diff
@@ -5,5 +5,5 @@
     "types": ["node", "cypress", "cypress-storybook/cypress", "cypress-real-events"],
     "lib": ["ES2019", "dom"]
   },
-  "include": ["**/*.cy.ts", "**/*.cy.tsx"]
+  "include": ["**/*.cy.ts", "**/*.cy.tsx", "./src/e2e/*.tsx"]
```
Comment (marcosmoura) [inline packages/react-components/react-drawer/tsconfig.cy.json:8]: This is to include the shared tests.

File: packages/react-components/react-drawer/src/e2e/DrawerShared.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+import { mount } from '@cypress/react';
+import { FluentProvider } from '@fluentui/react-provider';
+import { webLightTheme } from '@fluentui/react-theme';
+import { DrawerProps } from '@fluentui/react-drawer';
+
+const mountFluent = (element: JSX.Element) => {
+  mount(<FluentProvider theme={webLightTheme}>{element}</FluentProvider>);
+};
+
+export function testDrawerBaseScenarios(Component: React.ComponentType<DrawerProps>) {
```
Comment (marcosmoura) [inline packages/react-components/react-drawer/src/e2e/DrawerShared.tsx:11]: The shared tests in a function that can be called then for each base Drawer component.

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/b27e08e64320103ff77f1aed3ed05a592485acb9'>b27e08e64320103ff77f1aed3ed05a592485acb9</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29029/perf-test-react-components/Avatar-mount_base.html">609</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29029/perf-test-react-components/Avatar-mount.html">639</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29029/perf-test-react-components/Button-mount_base.html">297</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29029/perf-test-react-components/Button-mount.html">316</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29029/perf-test-react-components/Field-mount_base.html">1119</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29029/perf-test-react-components/Field-mount.html">1105</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29029/perf-test-react-components/FluentProvider-mount_base.html">692</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29029/perf-test-react-components/FluentProvider-mount.html">679</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29029/perf-test-react-components/FluentProviderWithTheme-mount_base.html">77</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29029/perf-test-react-components/FluentProviderWithTheme-mount.html">76</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29029/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">64</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29029/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">66</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29029/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">75</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29029/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">74</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29029/perf-test-react-components/InfoButton-mount_base.html">13</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29029/perf-test-react-components/InfoButton-mount.html">11</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29029/perf-test-react-components/MakeStyles-mount_base.html">858</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29029/perf-test-react-components/MakeStyles-mount.html">870</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29029/perf-test-react-components/Persona-mount_base.html">1743</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29029/perf-test-react-components/Persona-mount.html">1647</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29029/perf-test-react-components/SpinButton-mount_base.html">1285</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29029/perf-test-react-components/SpinButton-mount.html">1307</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by tudorpopams — APPROVED** [review]

**Pull request merged**